ivy/utils/backend/handler.py

```diff
@@ -236,9 +236,24 @@ def _handle_backend_specific_vars(target, backend):
         target.set_global_attr("RNG", target.functional.backends.jax.random.RNG)
 
 
-def convert_from_source_backend_to_numpy(variable_ids, numpy_objs, devices):
-    # Dynamic Backend
-    from ivy.functional.ivy.gradients import _is_variable, _variable_data
+def _data_to_new_backend(x, previous_backend):
+    device = previous_backend.dev(x)
+    try:
+        result = ivy.from_dlpack(previous_backend.to_dlpack(x.data))
+        result = ivy.to_device(result, device)
+    except Exception:
+        np_res = previous_backend.to_numpy(x.data)
+        result = ivy.asarray(np_res, device=device)
+    return result
+
+
+def dynamic_backend_converter(backend_stack):
+    from ivy.functional.ivy.gradients import _variable
+
+    if len(backend_stack) < 2:
```

Comment:
I think it's safe to assume that dynamic backend setting would only be used when one backend was set, a few arrays were created and then the user wants to switch to the second backend (i.e. `len(backend_stack)` would always be `>=2`).
Cuz we can't guarantee the backend of the array created when no backend was set (because any type of native array can be passed with implicit backend). And given that implicit backend is only supposed to be used for testing something out quickly, it makes our job much easier when all the arrays created were guaranteed to have the same backend.
Secondly, there might be some issues using a local `ivy` version (from `with_ backend`) and the global one, because the `ivy.Array` in the local version is different from the `ivy.Array` class in the global one, have you tested if the code works as expected?